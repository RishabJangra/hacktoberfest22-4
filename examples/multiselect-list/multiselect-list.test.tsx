import React from 'react';
import renderer from 'react-test-renderer';
import { MultiselectListScreen } from './MultiselectList';
import { ListItem } from './utilities';
import { InfoListItem } from '@pxblue/react-native-components';

jest.mock('@react-navigation/native', () => ({
    useNavigation: (): any => ({ openDrawer: jest.fn(() => true) }),
}));

jest.mock('react-native-safe-area-context', () => ({
    useSafeArea: (): any => ({}),
}));

const hardcodedData: ListItem[] = [
    {
        id: 1,
        name: 'item 1',
        details: 'item 1 details',
    },
    {
        id: 2,
        name: 'item 2',
        details: 'item 2 details',
    },
    {
        id: 3,
        name: 'item 3',
        details: 'item 3 details',
    },
];

describe('Multiselect List Tests', () => {
    it('Screen Renders', () => {
        const tree = renderer.create(<MultiselectListScreen hardcodedData={hardcodedData} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Should render 10 items by default', () => {
        const instance = renderer.create(<MultiselectListScreen />).root;
        const infoListItems = instance.findAllByType(InfoListItem);
        expect(infoListItems).toHaveLength(10);
    });
});

// @TODO implement these tests

// it('Should delete an Item', () => {

// });

// it('Should clear all Items', () => {

// });
