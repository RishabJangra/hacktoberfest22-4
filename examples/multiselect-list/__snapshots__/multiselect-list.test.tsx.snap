// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Multiselect List Tests Screen Renders 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    accessible={true}
    emulateUnlessSupported={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#6200ee",
        "elevation": 0,
        "height": 76,
        "shadowColor": "rgba(0, 0, 0, 0.3)",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 2,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "paddingBottom": 16,
          "paddingHorizontal": 16,
          "paddingTop": 16,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 40,
            "margin": -8,
            "marginRight": 24,
            "opacity": 1,
            "padding": 8,
            "width": 40,
          }
        }
        testID="header-navigation"
      >
        <Text />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "flex-end",
            "marginRight": 40,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 0,
              "justifyContent": "center",
            }
          }
        >
          <Text
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "color": "white",
                "fontFamily": "System",
                "fontSize": 20,
                "lineHeight": 20,
                "textAlign": "auto",
                "writingDirection": "ltr",
              }
            }
            testID="header-title"
          >
            Multiselect List
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 56,
              "position": "absolute",
              "right": 8,
            },
            undefined,
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "height": 40,
              "opacity": 1,
              "padding": 8,
              "width": 40,
            }
          }
          testID="header-action-item0"
        >
          <Text />
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "#ffffff",
        "flex": 1,
      }
    }
  >
    <RCTScrollView>
      <View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "flexDirection": "row",
              "height": 72,
              "opacity": 1,
              "paddingRight": 16,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": undefined,
                  "height": "100%",
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": 6,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "alignItems": "flex-start",
                  "justifyContent": "center",
                  "marginLeft": 16,
                  "width": 40,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "flex-start",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                    "width": 40,
                  },
                  undefined,
                ]
              }
            >
              <Text />
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "paddingHorizontal": 16,
                },
                undefined,
              ]
            }
          >
            <Text
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                  },
                  Object {
                    "writingDirection": "ltr",
                  },
                  Object {},
                  Object {
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "letterSpacing": 0,
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  undefined,
                  Array [
                    Object {
                      "color": "#000000",
                    },
                    undefined,
                  ],
                ]
              }
            >
              item 1
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                    },
                    Object {
                      "writingDirection": "ltr",
                    },
                    Object {},
                    Object {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0,
                    },
                    Object {
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    undefined,
                    Object {
                      "color": "#000000",
                    },
                  ]
                }
              >
                item 1 details
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "flexDirection": "row",
              "height": 72,
              "opacity": 1,
              "paddingRight": 16,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": undefined,
                  "height": "100%",
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": 6,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "alignItems": "flex-start",
                  "justifyContent": "center",
                  "marginLeft": 16,
                  "width": 40,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "flex-start",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                    "width": 40,
                  },
                  undefined,
                ]
              }
            >
              <Text />
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "paddingHorizontal": 16,
                },
                undefined,
              ]
            }
          >
            <Text
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                  },
                  Object {
                    "writingDirection": "ltr",
                  },
                  Object {},
                  Object {
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "letterSpacing": 0,
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  undefined,
                  Array [
                    Object {
                      "color": "#000000",
                    },
                    undefined,
                  ],
                ]
              }
            >
              item 2
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                    },
                    Object {
                      "writingDirection": "ltr",
                    },
                    Object {},
                    Object {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0,
                    },
                    Object {
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    undefined,
                    Object {
                      "color": "#000000",
                    },
                  ]
                }
              >
                item 2 details
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "flexDirection": "row",
              "height": 72,
              "opacity": 1,
              "paddingRight": 16,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": undefined,
                  "height": "100%",
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": 6,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "alignItems": "flex-start",
                  "justifyContent": "center",
                  "marginLeft": 16,
                  "width": 40,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "flex-start",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                    "width": 40,
                  },
                  undefined,
                ]
              }
            >
              <Text />
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "paddingHorizontal": 16,
                },
                undefined,
              ]
            }
          >
            <Text
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                  },
                  Object {
                    "writingDirection": "ltr",
                  },
                  Object {},
                  Object {
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "letterSpacing": 0,
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  undefined,
                  Array [
                    Object {
                      "color": "#000000",
                    },
                    undefined,
                  ],
                ]
              }
            >
              item 3
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                    },
                    Object {
                      "writingDirection": "ltr",
                    },
                    Object {},
                    Object {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0,
                    },
                    Object {
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    undefined,
                    Object {
                      "color": "#000000",
                    },
                  ]
                }
              >
                item 3 details
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
      </View>
    </RCTScrollView>
  </RCTSafeAreaView>
</View>
`;
